python/slurmTools.py

```diff
@@ -0,0 +1,407 @@
+
+        dico = {}
+        dico["#BASE_PATH#"] = self.baseDir
+        dico["#INDIR_PATH#"] = self.inDir
+        dico["#OUTDIR_PATH#"] = self.outDir
+        dico["#LOGDIR_RELPATH#"] = os.path.relpath(self.logDir)
+        dico["#EXEC_PATH#"] = self.execPath
+        dico["#PLOT_CFG_PATH#"] = self.plotConfig
+        dico["#CMS_PATH#"] = CMSSW_BASE
+
+        for sample in self.sampleCfg:
+            name = sample["db_name"]
+            jobList = sample["jobs"]
+
+            for job in jobList:
+
+                jsonFileName = os.path.join(self.inDir, "samples_{}.json".format(jobCount) )
+                jsonContent = copy.deepcopy(sample["json_skeleton"])
+                
+                if "#DB_NAME#" in jsonContent.keys():
+                    jsonContent = {}
+                    jsonContent[name] = copy.deepcopy(sample["json_skeleton"]["#DB_NAME#"])
+                    jsonContent[name]["db_name"] = name
+                
+                jsonContent[name]["files"] = job[0]
+                jsonContent[name]["event-start"] = job[1]
+                jsonContent[name]["event-end"] = job[2]
+                
+                if "output_name" not in jsonContent[name].keys():
+                    jsonContent[name]["output_name"] = name + "_histos"
+
+                with open(jsonFileName, "w") as js:
+                    json.dump(jsonContent, js)
+
+                jobCount += 1
+           
+        dico["#N_JOBS#"] = str(jobCount)
+
+        slurmFile = str(self.baseShell)
+
+        # self.baseCmd += self.jobCmd
+        for key in dico.items():
+            # self.baseCmd = self.baseCmd.replace(key[0], key[1])
+            slurmFile = slurmFile.replace(key[0], key[1])
+
+        # Shell wrapper
+        shFileName = os.path.join(self.inDir, "slurm.sh")
+        with open(shFileName, "w") as sh:
+            sh.write(slurmFile)
+        perm = os.stat(shFileName)
+        os.chmod(shFileName, perm.st_mode | stat.S_IEXEC)
+
+        print "Created {} job command files. Caution: the jobs are not submitted yet!.".format(jobCount)
+
+        self.jobCount = jobCount
+        self.isCreated = True
+
+        # Generate a command to hadd all the files when the jobs are done
+        haddFile = os.path.join(self.baseDir, "hadd_histos.sh")
+        with open(haddFile, "w") as f:
+            cmd = """
+#!/usr/bin/bash
+
+function check_success {
+  should_exit=false
+  find #LOGDIR_PATH# -name "*.err" -not -size 0 |
+  {
+    while read f; do
+      [[ $f =~ condor_([0-9]+) ]]
+      id=${BASH_REMATCH[1]}
+      echo "Error: job #${id} failed."
+      should_exit=true
+    done
+
+    if [ "${should_exit}" == true ]; then
+      exit 1
+    fi
+  }
+
+  if [ $? -eq 1 ]; then
+    exit 1
+  fi
+}
+
+function do_hadd {
+ if [ ! -z "$1" ]; then
+  hadd $1.root $1_*.root && if [ "$2" == "-r" ]; then rm $1_*.root; fi
+ fi
+}
+
+check_success
+
+base_list=`find -regex ".*_[0-9]*.root" -printf "%f\\n" | sed "s/_[0-9]*.root//g" | sort | uniq`
```

Comment:
The bug I was mentioning: this has to be executed after the `pushd` below... While you're at it can you fix it in condor as well or would you prefer I open a separate PR? :)